node_modules/oae-preview-processor/lib/processors/link/default.js

```diff
@@ -96,30 +102,60 @@ var generatePreviews = module.exports.generatePreviews = function(ctx, contentOb
              * All 3 of these would block embedding so as soon as the header is defined,
              * we add some metadata that tells the UI this link cannot be embedded.
              */
-            if (response.headers['x-frame-options']) {
+            if (response.headers['x-frame-options'] || forcedDownload) {
                 ctx.addPreviewMetadata('embeddable', false);
             } else {
                 ctx.addPreviewMetadata('embeddable', true);
             }
+
+            var contentType = response.headers['content-type'];
+            ctx.addPreviewMetadata('targetType', contentType);
         }
 
         /*!
          * Generate a thumbnail
          */
         var generateThumbnail = function() {
-            var path = ctx.baseDir + '/webshot.png';
-            // Try to localize the screenshot of the link to the default tenant language
-            webshotOptions.customHeaders = {
-                'Accept-Language': PrincipalsConfig.getValue(contentObj.tenant.alias, 'user', 'defaultLanguage')
-            };
-            webshot.getImage(contentObj.link, path, webshotOptions, function (err) {
-                if (err) {
-                    log().error({'err': err, 'contentId': ctx.contentId}, 'Could not generate an image');
-                    return callback(err);
-                }
-
-                LinkProcessorUtil.generatePreviewsFromImage(ctx, path, null, callback);
-            });
+            var imgPath = ctx.baseDir + '/webshot.png';
+            // If the link target is an image just grab it instead of screenshotting it
+            if (contentType && contentType.substring(0, 6) === 'image/') {
```

Comment:
...back when dinosaurs still roamed the earth. :smile: Do you think you'll have time to have a look at this or should I take over?